### PR TITLE
178318190 skipped vote median and average calculated in backround

### PR DESCRIPTION
**Pull request by maciejkocylapc**
#### What's this PR do?
reduce the load on the index page by moving skipped_vote calculations to validator_score_v1_logic

#### How should this be manually tested?
check if the numbers and charts are correct on the main page

#### What are the relevant tickets?
https://www.pivotaltracker.com/story/show/178318190

#### Task completed checklist:
- [Y] Is there appropriate test coverage?
- [Y] Did I take into consideration possible security vulnerabilities?
- [Y] Are the controllers secure?
- [Y] Was every important matter documented?

#### Things to watch out for
- [Y] Does this PR have any migrations?
- [N] Does this PR add new dependencies?

### CODE REVIEW

**Comment by radek-polcode** [comment]
undefined method `-' for nil:NilClass

 # Calculate the distance from the best skipped vote and round
      ((batch.best_skipped_vote - skipped_votes_percent.to_f) * 100.0).round(2)

`app/helpers/validators_helper.rb:61:in `skipped_vote_percent'`

File: app/views/validators/_spark_line_skipped_vote.html.erb [hunk 1]
```diff
@@ -5,9 +5,9 @@
         type: 'gauge',
         data: {
             datasets: [{
-                data: [-5, -3.0, 0.05],
+                data: [<%= -(@batch.skipped_vote_all_median * 20) %>, <%= -(@batch.skipped_vote_all_median * 10) %>, 0.05],
```
Comment (radek-polcode) [inline app/views/validators/_spark_line_skipped_vote.html.erb:8]: Why we multiplicate by 10 and 20 and by 30 below? Maybe you could create some meaningful method in the model to get rid of calculations in the view?

Comment (kbiala) [reply app/views/validators/_spark_line_skipped_vote.html.erb:8]: I don't think it should be placed in the model - it does not represent any data other than ranges in frontend chart.

Comment (kbiala) [reply app/views/validators/_spark_line_skipped_vote.html.erb:8]: Data is multiplicated for chart purposes only - first range is eqal median, second median*2, and so on.

Comment (radek-polcode) [reply app/views/validators/_spark_line_skipped_vote.html.erb:8]: However, please consider using some variables like - for someone who does not work on that these numbers are confusing.

Comment (kbiala) [reply app/views/validators/_spark_line_skipped_vote.html.erb:8]: @maciejkocylapc I'd pass them in render helper like account and value, I think that will be clear enough

File: app/helpers/validators_helper.rb [hunk 2]
```diff
@@ -53,7 +53,7 @@ def current_software_version(batch, network)
   end
 
   def skipped_vote_percent(validator, batch)
-    if validator.score&.skipped_vote_history
+    if validator.score&.skipped_vote_history && batch.best_skipped_vote
```
Comment (radek-polcode) [inline app/helpers/validators_helper.rb:56]: You can also use guard clause instead of if-else.

**Review by radek-polcode — APPROVED** [review]

**Pull request merged**